### ISSUE TITLE
[ENH] Add glossary to documentation

### ISSUE THREAD

**Issue by matusvalo** [cython/cython#4639]
I propose to add a glossary to cython documentation which will explain the terms used in documentation.

## Rationale

Cython is using plenty of concepts which should be explained in single place -e. g.

* cdef/@cfunc functions methods
* cpdef/@ccall functions or methods
* extention types / cdef classes
* cdef statement
* pyc files, pxd files ...
* ...

See also the discussion here: https://github.com/cython/cython/pull/4516#discussion_r799387386

## Implementation

We should do something similar what is python doing: https://docs.python.org/3/glossary.html

**Comment by da-woods** [comment]
We added a glossary already https://github.com/cython/cython/blob/master/docs/src/userguide/glossary.rst.

It would be good to add to it, and get more links to it though

**Comment by matusvalo** [comment]
Hmm I missed it. It is kind of buried in documentation. I have checked the main page of documentation (searched for glossary) and it was not mentioned there - it is in userguide at the bottom. E.g. python on the other hand has it pretty visible - see screenshot below. But also links to the defined names would be great and would help a lot. Let me rephrase the issue.

![image](https://user-images.githubusercontent.com/827060/153775846-3a73b380-f379-4bed-a082-e03f170eaf93.png)

**Comment by scoder** [comment]
> https://github.com/cython/cython/blob/master/docs/src/userguide/glossary.rst.

I hadn't even noticed that it's hidden under `userguide/`. That makes no sense. I'd rather move the file to the top, but then we'd lose the links to it. So I moved at least the link to the main index page. https://github.com/cython/cython/commit/f581ff4f33b9a65a514a580c78fd944e92e83ae2
